dialogflow/stopwatch/Stopwatch/StopwatchService.swift

```diff
@@ -38,121 +36,89 @@ enum StopwatchServiceError: Error {
-      inputAudioConfig.audioEncoding = DFAudioEncoding(rawValue:1)!
-      inputAudioConfig.languageCode = "en-US"
-      inputAudioConfig.sampleRateHertz = Int32(sampleRate)
-      queryInput.audioConfig = inputAudioConfig
+    // authenticate using an authorization token (obtained using OAuth)
+    authorization { (authT) in
+      if (!self.streaming) {
+        // if we aren't already streaming, set up a gRPC connection
+        self.client = DFSessions(host:Host)
+        self.writer = GRXBufferedPipe()
+        self.call = self.client.rpcToStreamingDetectIntent(
+          withRequestsWriter: self.writer,
+          eventHandler: { (done, response, error) in
+            completion(response, error as NSError?)
+        })
+        self.call.requestHeaders.setObject(NSString(string:authT),
+                                           forKey:NSString(string:"Authorization"))
+        self.call.start()
+        self.streaming = true
+
+        // send an initial request message to configure the service
+
+        let queryInput = DFQueryInput()
+        let inputAudioConfig = DFInputAudioConfig()
+        inputAudioConfig.audioEncoding = DFAudioEncoding(rawValue:1)!
+        inputAudioConfig.languageCode = "en-US"
+        inputAudioConfig.sampleRateHertz = Int32(self.sampleRate)
+        queryInput.audioConfig = inputAudioConfig
+
+        let streamingDetectIntentRequest = DFStreamingDetectIntentRequest()
+        streamingDetectIntentRequest.session = "projects/" + ProjectName +
+          "/agent/sessions/" + SessionID
+        streamingDetectIntentRequest.singleUtterance = true
+        streamingDetectIntentRequest.queryParams = self.getQueryParmasFor()
+        streamingDetectIntentRequest.queryInput = queryInput
+        streamingDetectIntentRequest.outputAudioConfig = self.getOutputAudioConfig()
+        self.writer.writeValue(streamingDetectIntentRequest)
+      }
 
+      // send a request message containing the audio data
       let streamingDetectIntentRequest = DFStreamingDetectIntentRequest()
-      streamingDetectIntentRequest.session = "projects/" + ProjectName +
-        "/agent/sessions/" + SessionID
-      streamingDetectIntentRequest.singleUtterance = true
-      streamingDetectIntentRequest.queryParams = getQueryParmasFor()
-      streamingDetectIntentRequest.queryInput = queryInput
-      streamingDetectIntentRequest.outputAudioConfig = getOutputAudioConfig()
-      writer.writeValue(streamingDetectIntentRequest)
+      streamingDetectIntentRequest.inputAudio = audioData as Data
+      self.writer.writeValue(streamingDetectIntentRequest)
     }
-
-    // send a request message containing the audio data
-    let streamingDetectIntentRequest = DFStreamingDetectIntentRequest()
-    streamingDetectIntentRequest.inputAudio = audioData as Data
-    writer.writeValue(streamingDetectIntentRequest)
   }
 
   func streamText(_ userInput: String, completion: @escaping StopwatchTextCompletionHandler) {
```

Comment:
We aren't quite streaming text in this method, just sending a single text request. Let's rename to `sendText` so as not to confuse folks.

Reply:
Yeah will do that.